homeassistant/components/weatherkit/weather.py

```diff
@@ -0,0 +1,255 @@
+        "native_wind_gust_speed": forecast.get("windGust"),
+        "native_wind_speed": forecast.get("windSpeed"),
+        "wind_bearing": forecast.get("windDirection"),
+        "humidity": forecast.get("humidity") * 100,
+        "native_precipitation": forecast.get("precipitationAmount"),
+        "precipitation_probability": forecast.get("precipitationChance") * 100,
+        "cloud_coverage": forecast.get("cloudCover") * 100,
+        "uv_index": forecast.get("uvIndex"),
+    }
+
+
+class WeatherKitWeather(
+    SingleCoordinatorWeatherEntity[WeatherKitDataUpdateCoordinator]
+):
+    """Weather entity for Apple WeatherKit integration."""
+
+    _attr_attribution = ATTRIBUTION
+
+    _attr_has_entity_name = True
+    _attr_native_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_native_pressure_unit = UnitOfPressure.MBAR
+    _attr_native_visibility_unit = UnitOfLength.KILOMETERS
+    _attr_native_wind_speed_unit = UnitOfSpeed.KILOMETERS_PER_HOUR
+
+    _attr_supported_features = (
+        WeatherEntityFeature.FORECAST_DAILY | WeatherEntityFeature.FORECAST_HOURLY
+    )
+
+    def __init__(
+        self,
+        coordinator: WeatherKitDataUpdateCoordinator,
+        config: MappingProxyType[str, Any],
+    ) -> None:
+        """Initialise the platform with a data instance and site."""
+        super().__init__(coordinator)
+        self._config = config
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID."""
+        location = self._config[CONF_LOCATION]
+        return f"{location[CONF_LATITUDE]}-{location[CONF_LONGITUDE]}"
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        name: str = self._config.get(CONF_NAME, "Apple Weather")
+        return name
+
+    @property
+    def condition(self) -> str | None:
+        """Return the current condition."""
+        condition_code = self.coordinator.data.get("currentWeather").get(
+            "conditionCode"
+        )
+        condition = condition_code_to_hass[condition_code]
+
+        if (
+            condition == "sunny"
+            and self.coordinator.data.get("currentWeather").get("daylight") is False
+        ):
+            condition = "clear-night"
+
+        return condition
+
+    @property
+    def native_temperature(self) -> float | None:
+        """Return the current temperature."""
+        temperature = self.coordinator.data.get("currentWeather").get("temperature")
+        return temperature
+
+    @property
+    def native_apparent_temperature(self) -> float | None:
+        """Return the current apparent_temperature."""
+        apparent_temperature = self.coordinator.data.get("currentWeather").get(
+            "temperatureApparent"
+        )
+        return apparent_temperature
+
+    @property
+    def native_dew_point(self) -> float | None:
+        """Return the current dew_point."""
+        dew_point = self.coordinator.data.get("currentWeather").get(
```

Comment:
You could make a 

```
@property
def data(self) -> `Type here`:
    """Return coordinator data."""
    return self.coordinator.data

Than you can replace all the `self.coordinator.data` calls with `self.data`

Its a tiny bit slower but makes all the properties a lot more readable

Reply:
You might also want something that like this to avoid writing out all the currentWeather gets

```python
@property
def current_weather(self) -> dict[str, Any]:
    """Return current weather data."""
    return self.coordinator.data.get("currentWeather")
```